### ISSUE TITLE
Support variable framerate in libTAS movies

### ISSUE THREAD

**Issue by InfoTeddy** [TASVideos/tasvideos#808]
As reported by duuuuude5, movies with variable framerate (e.g. a framerate that can be different on each frame) will have their length displayed inaccurately, because the site calculates the length by taking the singular framerate and dividing it by the number of frames, instead of taking into account the framerate of each frame.

This can be fixed by simply using the length field in the movie file. libTAS stores the length of the movie file for the exact reason of avoiding calculating the length manually.

**Comment by vadosnaprimer** [comment]
Please attach a movie that highlights this.

**Comment by InfoTeddy** [comment]
[7242S](https://tasvideos.org/7242S), which was where duuuuude5 noticed this.

**Comment by CasualPokePlayer** [comment]
> [7242S](https://tasvideos.org/7242S), which was where duuuuude5 noticed this.

Note the time here would be an import time (and matches exactly with the old site), which would not go through the same process as newer movies. Although re-submitting that locally does show the exact same time.

Interestingly, the length_nsec and length_sec are the same value for some reason? Along with the value having a nonsense value at around 584 years, and that's assuming it's parsed as nanoseconds. How would this even occur?

**Comment by adelikat** [comment]
So is this an error with the file?

**Comment by Masterjun3** [comment]
The issue here is that libTAS implemented a variable framerate feature and the parser isn't prepared for this. The framerate field in the movie file is only the initial framerate value. This means we have one of three options:
1. Disallow variable framerates completely.
2. Use libTAS's own recorded movie length in the movie file if variable framerates are enabled.
3. Ask the libTAS dev to add an "average framerate" field to the movie file.

**Comment by vadosnaprimer** [comment]
> [7242S](https://tasvideos.org/7242S), which was where duuuuude5 noticed this.

What time is it meant to be? I see nothing in libtas

![изображение](https://user-images.githubusercontent.com/7092625/150654481-7a7822c4-1ce5-4c24-9f6a-4adee4cb32fc.png)

**Comment by InfoTeddy** [comment]
I'm not quite sure, it looks like the variable framerate information got lost somehow. I think we should ask duuuuude5.

**Comment by vadosnaprimer** [comment]
Here's the fixed movie and it's length is 1:22:29.95 (I needed to export the movie on the next frame after it pauses at the end).

Relevant fields in `config.ini`:

```
length_nsec=950999990
length_sec=4949
variable_framerate=true
```

**Comment by vadosnaprimer** [comment]
Renamed the issue. We need this because libTAS has this feature, and movie time won't be shown properly if we don't use movie duration from the header. There's at least one current submission that needs this, and potentially indefinite number of new submissions too.
